src/datastores/handlers/base.js

```diff
@@ -167,12 +167,39 @@ class Playlists {
   }
 }
 
+class SearchHistory {
```

Comment:
Probably should rename all related code to page bookmark?

Reply:
The intention is that we could have persistent keyword search history entry logic here in the future (see `Future PRs` in OP), and page bookmarks are a type of search history.